src/commands/local.ts

```diff
@@ -343,6 +363,138 @@ export async function startLocalEnvironment(options: GenezioLocalOptions) {
             eventType: TelemetryEventTypes.GENEZIO_LOCAL_RELOAD,
             commandOptions: JSON.stringify(options),
         });
+        clearTimeout(nodeModulesFolderWatcher);
+    }
+}
+
+async function watchNodeModules(
+    yamlProjectConfiguration: YamlProjectConfiguration,
+    sdkPath: string,
+): Promise<NodeJS.Timeout> {
+    // We are watching for the following files:
+    // - node_modules/@genezio-sdk/<projectName>_<region>/package.json: this file is used to determine if the SDK was changed (by a npm install or npm update)
+    // - node_modules/.package-lock.json: this file is used by npm to determine if it should update the packages or not. We are removing this file while "genezio local"
```

Comment:
Keep in mind that this is incompatible with yarn and pnpm. You can leave a TODO comment if you want to approach this later. 

Pnpm is using a `.pnpm` file and yarn is using a `.yarn.state` file.